data/data/powervs/bootstrap/vm/main.tf

```diff
@@ -50,7 +50,8 @@ resource "ibm_pi_instance" "bootstrap" {
     network_id = var.dhcp_network_id
   }
   pi_user_data = base64encode(templatefile("${path.module}/templates/bootstrap.ign", {
-    HOSTNAME    = ibm_cos_bucket.ignition.s3_endpoint_public
+    PROTOCOL    = ! var.enable_snat ? "http" : "https"
+    HOSTNAME    = ! var.enable_snat ? var.proxy_server_ip : ibm_cos_bucket.ignition.s3_endpoint_public
```

Comment:
Negative condition confuses me a little bit, but maybe only me?

Reply:
Not the nicest looking thing but intent is that the parameter it will set is `pi_dhcp_enable_snat` so matches that. Negated here because we want to use the proxy VM for COS when SNAT is disabled (! enable_snat)